components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AddonCollectionProvider.kt

```diff
@@ -370,7 +418,14 @@ internal fun JSONObject.getSafeJSONArray(key: String): JSONArray {
         getJSONArray(key)
     }
 }
-
+internal fun JSONObject.getSafeTranslations(valueKey: String, language: String?): Map<String, String> {
+    return if (get(valueKey) is String) {
+        val safeLanguage = language ?: Addon.DEFAULT_LOCALE
+        mapOf(safeLanguage to getSafeString(valueKey))
+    } else {
+        getSafeMap(valueKey)
+    }
+}
```

Comment:
@csadilek the [get](https://android.googlesource.com/platform/dalvik/+/f2f7880a40aed1c8d9c27db49226e47396556aac/libcore/json/src/main/java/org/json/JSONObject.java#325) function returns an `Object`  of the key provided, when the JSON representation is a`String` it returns a `String` and when the JSON representation is an `Object` it returns a `JSONObject` and this behavior make it ideal for what we want to achieve here.

Reply:
nice, that looks good!